### PR TITLE
docs: add contributing

### PR DESCRIPTION
**Pull request by hanyuancheung**
feat: add contributing md

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,512 @@
+# Contributing to opentelemetry-go-build-tools
+
+The Go special interest group (SIG) meets regularly. See the
```
Comment (pellared) [inline CONTRIBUTING.md:3]: We do not have any dedicated SIG for Go-build tools. I think what we have in `README.md` is good enough.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,512 @@
...
+understand, but `git` does not.)
+
+This would put the project in the `opentelemetry-go-build-tools` directory in
+current working directory.
+
+Enter the newly created directory and add your fork as a new remote:
+
+```sh
+git remote add <YOUR_FORK> git@github.com:<YOUR_GITHUB_USERNAME>/opentelemetry-go-build-tools
+```
```
Comment (pellared) [inline CONTRIBUTING.md:70]: This does not look correct (especially `go get -d go.opentelemetry.io/otel`). I think we can remove this part and simply say that one needs to work on a fork.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,512 @@
...
+and the `Close` method called.
+
+This super-set approach can be useful if there is explicit behavior that needs
+to be coupled with the original type and passed as a unified type to a new
+function, but, because of this coupling, it also limits the applicability of
+the added functionality. If there exist other interfaces where this
+functionality should be added, each one will need their own super-set
+interfaces and will duplicate the pattern. For this reason, the simple targeted
+interface that defines the specific functionality should be preferred.
+
```
Comment (pellared) [inline CONTRIBUTING.md:500]: This should be removed.

**Comment by pellared** [comment]
@hanyuancheung Are you planning to follow up on this PR?

**Comment by hanyuancheung** [comment]
> @hanyuancheung Are you planning to follow up on this PR?

Sure, I'll fix the comment above.

**Comment by hanyuancheung** [comment]
I've fixed, PTAL again, thanks. @pellared

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,346 @@
+# Contributing to opentelemetry-go-build-tools
+
+See the [public meeting
+notes](https://docs.google.com/document/d/1A63zSWX0x2CyCK_LoNhmQC4rqhLpYXJzXbEPDUQ2n6w/edit#heading=h.9tngw7jdwd6b)
+for a summary description of past meetings. To request edit access,
+join the meeting or get in touch on
+[Slack](https://cloud-native.slack.com/archives/C01NPAXACKT).
```
Comment (pellared) [inline CONTRIBUTING.md:7]: This should be removed.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+}
+```
+
+## Approvers and Maintainers
+
+CodeOwners:
+
+* @open-telemetry/go-maintainers
+* @open-telemetry/collector-maintainers
+* @open-telemetry/collector-contrib-approvers
```
Comment (pellared) [inline CONTRIBUTING.md:341]: This list is not up to date. Maybe simply

```suggestion
[CODEOWNERS](https://github.com/open-telemetry/opentelemetry-go-build-tools/blob/main/.github/CODEOWNERS)
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+}
+
+type weightOption float64
+
+func (o weightOption) applyDog(c config) config {
+c.Weight = float64(o)
+return c
+}
+```
+
```
Comment (pellared) [inline CONTRIBUTING.md:334]: This is not applicable and can be removed

```suggestion
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,346 @@
+# Contributing to opentelemetry-go-build-tools
```
Comment (pellared) [inline CONTRIBUTING.md:1]: Can you link to this file in `README.md`?

Comment (hanyuancheung) [reply CONTRIBUTING.md:1]: Done

**Review by pellared — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Review by XSAM — APPROVED** [review]

**Pull request merged**